### ISSUE TITLE
Add support for Artifactory projects

### ISSUE THREAD

**Issue by mirath** [jfrog/terraform-provider-artifactory#74]
The new [projects](https://www.jfrog.com/confluence/display/JFROG/Projects) feature was released recently and we are very interested in using it. However going over the docs [here](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs) it seems that there is not support yet.

**Comment by chb0github** [comment]
@markgalpin I have marked this as a feature request

**Comment by mdaalman** [comment]
I agree, I would like to see this integrated in the terraform jfrog artifactory provider as well.

**Comment by rhooverea** [comment]
I agree as well, projects support would be super helpful

**Comment by markgalpin** [comment]
We are definitely considering this as a feature for next quarter.  If you have any more detailed specifics on what projects use cases are most important for you in the terraform provider, that will help us prioritize it.

**Comment by mmclane** [comment]
I want to be able to create and configure projects using terraform.  Additionally, I want to be able to create an configure repositories within a project

**Comment by advissor** [comment]
+1 
We would like to Create Projects, create application users and attach roles + probably create repositories under this project
So, all the resources would be are scoped to certain project

**Comment by mickael1026** [comment]
We would like to be able to manage project role and global role.
Having a data sources project will also help us. Jfrog admin will create the project then the project admin should be able to manage an existing project in an other terraform workspace.
Data sources for group, global role will also help us.

**Comment by mmclane** [comment]
For now I have a working solution using the restapi resource.  Its not early as pretty but it works.

**Comment by ggolub** [comment]
+1
We just got started with this and cannot create repos because we are restricted by our IT to only being able to do so within projects.  Since we can't specify a project key we are blocked.

**Comment by chb0github** [comment]
Support for projects was evaluated a couple of months ago and there was no way to get the list of repos in a project. Ergo, we couldn't implement it. It didn't even work properly in the UI. I just checked, and it seems you can now get the list of repos for a project, by supplying the projectKey as a query param when listing repositories

`curl -snL "https://partnership.jfrog.io/artifactory/api/repositories?projectKey=foo"`

An important note: the new endpoints that were added for project APIs _ONLY_  support `Authorization: Bearer` (the ones in the admin section), and those could not be scoped down to `projects` and thus, require admin to function. So, the only way to make this work, currently, is to give out a god token

**Comment by Natanande** [comment]
> Please vote for https://www.jfrog.com/jira/browse/RTFACT-26289 which should provide the functionality to allow this feature to be added to the provider.

It seems like they have resolved this issue now!
